src/adaptix/_internal/morphing/enum_provider.py

```diff
@@ -155,3 +210,97 @@ def _get_exact_value_to_member(self, enum: Type[Enum]) -> Optional[Mapping[Any,
 
     def _provide_dumper(self, mediator: Mediator, request: DumperRequest) -> Dumper:
         return _enum_exact_value_dumper
+
+
+def _extract_non_compound_cases_from_flag(enum: Type[FlagT]) -> Iterable[FlagT]:
+    return [case for case in enum.__members__.values() if not math.log2(case.value) % 1]
+
+
+class FlagProvider(BaseEnumProvider):
+    def __init__(
+        self,
+        mapping_generator: BaseEnumMappingGenerator,
+        allow_single_value: bool = False,
+        allow_duplicates: bool = True,
+        allow_compound: bool = True,
+    ):
+        self._mapping_generator = mapping_generator
+        self._allow_single_value = allow_single_value
+        self._allow_duplicates = allow_duplicates
+        self._allow_compound = allow_compound
+
+    def _get_cases(self, enum: Type[FlagT]) -> Iterable[FlagT]:
+        if self._allow_compound:
+            return enum.__members__.values()
+        return _extract_non_compound_cases_from_flag(enum)
+
+    @overload
+    def _get_loader_process_data(self, data: Union[int, Iterable[int]], enum: Type[Flag]) -> Sequence[int]:
+        ...
+
+    @overload
+    def _get_loader_process_data(self, data: Iterable[str], enum: Type[Flag]) -> Sequence[str]:
+        ...
+
+    def _get_loader_process_data(self, data, enum):
+        if isinstance(data, (str, int)):
+            if not self._allow_single_value:
+                raise TypeLoadError(expected_type=Iterable[str], input_value=data)
+            process_data = [data]
+        else:
+            process_data = list(data)
+
+        if not self._allow_duplicates:
+            if len(process_data) != len(set(process_data)):
+                raise ValueLoadError(
+                    msg=f"Duplicates in {enum} loader are not allowed",
```

Comment:
API user knows nothing about what loader and enum class is it. Also I think it should be separate error class like 
```python

@custom_exception
@dataclass(eq=False)
class DuplicatedValues:
    input_value: Any
```